packages/keyring-controller/src/KeyringController.test.ts

```diff
@@ -502,7 +501,8 @@ describe('KeyringController', () => {
     );
     const recovered = recoverTypedSignature({
       data: msgParams as any,
```

Comment:
What do you think about removing this typecasting? It looks like this can be done by changing `primaryType: 'Mail'` above to `primaryType: 'Mail' as const`.

Reply:
Why does this work?

Reply:
done [here](https://github.com/MetaMask/controllers/pull/914/commits/72149b509d1e174c837ce9e227568b9a52582a5f)

Reply:
`recoverTypedSignature` appears to infer the expected type of `data` based on the version you provide. In this case it wants the data to be [`TypedMessage<T>`](https://github.com/MetaMask/eth-sig-util/blob/a7a67795331714c6f0d841e3a71c50a76f85823a/src/sign-typed-data.ts#L81-L92). This type says that `types` is `T` and `primaryType` should be a key of `T`. It doesn't really matter what `T` is in this case — the point is that TypeScript wants to unify `primaryType` with a key of `types` (i.e. `"EIP712Domain" | "Mail" | "Person"`). However, the type of `primaryType` in this case is actually `string` and not one of those three known strings. By adding `as const` after "Mail", we're telling TypeScript to use the value of the string as its own type. That makes the type of `msgParams` internally consistent and compatible with `TypedMessage<T>`.